Primo.js

```diff
@@ -314,12 +324,22 @@ function importPNX(text) {
 	}
 	
 	// the three letter ISO codes that should be in the language field work well:
-	item.language = ZU.xpathText(doc, '//display/language');
+	item.language = ZU.xpathText(doc, '//display/language|//facets/language');
 	
+	// BEIC.it uses //dedup/f9, unclear what that is
 	var pages = ZU.xpathText(doc, '//display/format');
 	if(pages && pages.search(/[0-9]+/) != -1) {
-		pages = pages.replace(/[\(\)\[\]]/g, "").match(/[0-9]+/);
-		item.pages = item.numPages = pages[0];
+		pages = pages.replace(/[\(\)\[\]]/g, "");
+		// And then we'd hope pages are the only number mentioned in the format
+		// Except that they aren't, in BEIC.it, so we focus the match a bit
+		if(pages.search(/ p+[. ]/) != -1) {
+			var dirt = pages;
+			pages = null;
+			pages = dirt.match(/(?:p+[. ]*)([0-9]+) *(?:p+[. ]*)/);
```

Comment:
The regexp requires a `p` in the beginning <b>and</b> in the end? That seems wrong for me and it seems not to work in your test case.